addons/mail/static/src/models/messaging_notification_handler.js

```diff
@@ -186,6 +192,7 @@ registerModel({
          * @param {boolean} [payload.open_chat_window] if true, will pin the channel
          */
         _handleNotificationChannelJoined({ channel: channelData, invited_by_user_id: invitedByUserId, open_chat_window: openChatWindow }) {
+            this.env.services.bus_service.updateChannels();
```

Comment:
Considering the number of occurrences, shouldn't it be part of the channel model instead? -> whenever the is pinned state of the channel changes, this should be called?

Reply:
Indeed, I believe putting this line in `_onIsServerPinnedChanged` is what we want, it appears to work well.